src/libraries/System.Private.CoreLib/src/System/Collections/HashHelpers.cs

```diff
@@ -90,21 +90,23 @@ public static int ExpandPrime(int oldSize)
         }
 
 #if BIT64
+        // Returns 64-bit multiplicative inverse of the divisor: ceil(2**64 / divisor)
         public static ulong GetFastModMultiplier(uint divisor)
             => ulong.MaxValue / divisor + 1;
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static uint FastMod(uint value, uint divisor, ulong multiplier)
         {
-            // Using fastmod from Daniel Lemire https://lemire.me/blog/2019/02/08/faster-remainders-when-the-divisor-is-a-constant-beating-compilers-and-libdivide/
+            // Using improved Daniel Lemire's fastmod: https://github.com/dotnet/runtime/pull/406
```

Comment:
This should probably stay linking to the original location that the algorithm came from.

Reply:
I have added the full proof of correctness of this algorithm to this PR. It is short and it does not require reading the original paper.

Reply:
Right, but linking to this PR doesn't really help people interested in the original paper to find it. It just adds another layer of indirection to finding https://lemire.me/blog/2019/02/08/faster-remainders-when-the-divisor-is-a-constant-beating-compilers-and-libdivide/ or https://github.com/lemire/fastmod.

There isn't any harm in keeping the original/longer URL but the new comment makes it seem like Lemire's fastmod can be found at the URL listed and it can't.

Reply:
> the new comment makes it seem like Lemire's fastmod can be found at the URL listed and it can't.

@tannergooding We are no longer using the original LKK's fastmod. Instead we use a _different_ implementation described in this PR — not in the original paper.

> It just adds another layer of indirection to finding

That blog post does not help you to understand why our algorithm works.  This PR, however, does.  One may completely skip the original paper and read just this PR description, which considers only one specific case (32-bit unsigned integers), and therefore much shorter.

Reply:
>  This PR, however, does. 

Ah, the summary of PR is a mini-paper, as it were